### PR TITLE
Feature/disjoint smart graphs fix query snippets

### PR DESCRIPTION
**Pull request by hkernbach**

### CODE REVIEW

File: arangod/Aql/QuerySnippet.cpp [hunk 1]
```diff
@@ -403,11 +403,14 @@ ResultT<std::unordered_map<ExecutionNode*, std::set<ShardID>>> QuerySnippet::pre
           // OneShard case) because local graphs (on DBServers) only ever occur
           // in either OneShard or SatelliteGraphs.
```
Comment (markuspf) [inline arangod/Aql/QuerySnippet.cpp:404]: Comment should be updated to reflect that disjoint smart graphs can occur here too.

File: arangod/Aql/QuerySnippet.cpp [hunk 2]
```diff
@@ -403,11 +403,14 @@ ResultT<std::unordered_map<ExecutionNode*, std::set<ShardID>>> QuerySnippet::pre
...
           TRI_ASSERT(found->second == server || localGraphNode->isUsedAsSatellite() ||
-                     aqlCollection->isSatellite());
+                     aqlCollection->isSatellite() || localGraphNode->isDisjoint());
           // provide a correct translation from collection to shard
           // to be used in toVelocyPack methods of classes derived
           // from GraphNode
-          localGraphNode->addCollectionToShard(aqlCollection->name(), shard);
+          if (found->second == server || !localGraphNode->isDisjoint()) {
+            // TODO: Optimize this, we're looking into to many collections here
```
Comment (markuspf) [inline arangod/Aql/QuerySnippet.cpp:411]: *when* do we optimize this? Do we have at least a Kanban card for it?

File: arangod/Aql/QuerySnippet.cpp [hunk 3]
```diff
@@ -418,10 +421,12 @@ ResultT<std::unordered_map<ExecutionNode*, std::set<ShardID>>> QuerySnippet::pre
         THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL_AQL, "Could not find a shard to instantiate for graph node when expected to");
       }
 
-      auto foundEnoughShards = numShards == localGraphNode->collections().size();
-      TRI_ASSERT(foundEnoughShards);
-      if (!foundEnoughShards) {
-        THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL_AQL);
+      if (localGraphNode->isEligibleAsSatelliteTraversal()) {
```
Comment (markuspf) [inline arangod/Aql/QuerySnippet.cpp:424]: Maybe this should be renamed to `isEligibleAsLocalTraversal`?

File: arangod/Aql/QuerySnippet.cpp (hunk 2 above)
Comment (hkernbach) [reply arangod/Aql/QuerySnippet.cpp:411]: Yes we have

**Pull request merged**